torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1637,6 +1639,14 @@ def sample_inputs_masked_select(op_info, device, dtype, requires_grad):
            sample_inputs_func=sample_inputs_fliplr_flipud,
            test_inplace_grad=False,
            supports_tensor_out=False),
+    UnaryUfuncInfo('i0',
+                   ref=np.i0,
```

Comment:
the i0 tests are really interesting

as we start to pursue more special function it might be helpful to think about how tests like the one i0 is doing could be made easier to write, possibly by extending UnaryUfuncInfo (or creating a subclass of it, SpecialInfo)

Reply:
Yes. The actual tests verifies its behaviour against the reference in multiple ranges.

https://github.com/pytorch/pytorch/blob/0620c96fd6a140e68c49d68ed14721b1ee108ecc/test/test_unary_ufuncs.py#L1035-L1066

UnaryUfunc suite tests following values (all of the ranges for i0 are not verified),

https://github.com/pytorch/pytorch/blob/0620c96fd6a140e68c49d68ed14721b1ee108ecc/test/test_unary_ufuncs.py#L43-L61

It would be interesting to extend making such things easier. But I guess we can take more informed decision after looking at other operators as well.